src/Renderer/ThreeExtended/FirstPersonControls.js

```diff
@@ -59,22 +111,59 @@ const MOVEMENTS = {
 
 
 class FirstPersonControls extends THREE.EventDispatcher {
+    /**
+     * @Constructor
+     * @param {View} view
+     * @param {object} options
+     * @param {boolean} options.focusOnClick - whether or not to focus the renderer domElement on click
+     * @param {boolean} options.focusOnMouseOver - whether or not to focus when the mouse is over the domElement
+     * @param {boolean} options.moveSpeed - if > 0, pressing the arrow keys will move the camera
+     * @param {number} options.verticalFOV - define the max visible vertical angle of the scene in degrees (default 180)
+     * @param {number} options.panoramaRatio - alternative way to specify the max vertical angle when using a panorama.
+     * You can specify the panorama width/height ratio and the verticalFOV will be computed automatically
+     */
     constructor(view, options = {}) {
         super();
         this.camera = view.camera.camera3D;
         this.view = view;
         this.moves = new Set();
-        this.moveSpeed = options.moveSpeed || 10; // backward or forward move speed in m/s
-        this._isUserInteracting = false;
+        if (options.panoramaRatio) {
+            const radius = (options.panoramaRatio * 200) / (2 * Math.PI);
+            options.verticalFOV =
+                options.panoramaRatio == 2 ? 180 : THREE.Math.radToDeg(2 * Math.atan(200 / (2 * radius)));
```

Comment:
`options.panoramaRatio == 2` is a state?

Reply:
Nope. 2 is the image ratio for equirectangular panoramas.